src/dock.js

```diff
@@ -0,0 +1,739 @@
+      // Never animate toggling visiblity...
+      nextState.shouldAnimate = false
+    } else if (!nextState.open && nextState.draggingItem && !prevState.draggingItem) {
+      // ...but do animate if you start dragging while the panel is hidden.
+      nextState.shouldAnimate = true
+    }
+
+    this.state = nextState
+    this.render(this.state)
+  }
+
+  render (state) {
+    if (this.element == null) {
+      this.element = document.createElement('atom-dock')
+      this.element.classList.add(this.location)
+      this.innerElement = document.createElement('div')
+      this.innerElement.classList.add('atom-dock-inner', this.location)
+      this.maskElement = document.createElement('div')
+      this.maskElement.classList.add('atom-dock-mask')
+      this.wrapperElement = document.createElement('div')
+      this.wrapperElement.classList.add('atom-dock-content-wrapper', this.location)
+      this.resizeHandle = new DockResizeHandle({
+        location: this.location,
+        onResizeStart: this.handleResizeHandleDragStart,
+        toggle: this.toggle.bind(this)
+      })
+      this.toggleButton = new DockToggleButton({
+        onDragEnter: this.handleToggleButtonDragEnter.bind(this),
+        location: this.location,
+        toggle: this.toggle.bind(this)
+      })
+      this.cursorOverlayElement = document.createElement('div')
+      this.cursorOverlayElement.classList.add('atom-dock-cursor-overlay', this.location)
+
+      // Add the children to the DOM tree
+      this.element.appendChild(this.innerElement)
+      this.innerElement.appendChild(this.maskElement)
+      this.maskElement.appendChild(this.wrapperElement)
+      this.wrapperElement.appendChild(this.resizeHandle.getElement())
+      this.wrapperElement.appendChild(this.paneContainer.getElement())
+      this.wrapperElement.appendChild(this.cursorOverlayElement)
+      // The toggle button must be rendered outside the mask because (1) it shouldn't be masked and
+      // (2) if we made the mask larger to avoid masking it, the mask would block mouse events.
+      this.innerElement.appendChild(this.toggleButton.getElement())
+    }
+
+    if (state.open) {
+      this.innerElement.classList.add(OPEN_CLASS)
+    } else {
+      this.innerElement.classList.remove(OPEN_CLASS)
+    }
+
+    if (state.shouldAnimate) {
+      this.maskElement.classList.add(SHOULD_ANIMATE_CLASS)
+    } else {
+      this.maskElement.classList.remove(SHOULD_ANIMATE_CLASS)
+    }
+
+    if (state.resizing) {
+      this.cursorOverlayElement.classList.add(CURSOR_OVERLAY_VISIBLE_CLASS)
+    } else {
+      this.cursorOverlayElement.classList.remove(CURSOR_OVERLAY_VISIBLE_CLASS)
+    }
+
+    const shouldBeVisible = state.open || state.showDropTarget
+    const size = Math.max(MINIMUM_SIZE, state.size == null ? this.getInitialSize() : state.size)
+
+    // We need to change the size of the mask...
+    this.maskElement.style[this.widthOrHeight] = `${shouldBeVisible ? size : HANDLE_SIZE}px`
```

Comment:
The `4px` gap covers these git markers when a dock is collapsed.

Open | Collapsed
--- | ---
![screen shot 2017-03-25 at 8 20 12 pm](https://cloud.githubusercontent.com/assets/378023/24321811/cb3b5e4e-1198-11e7-915f-43df25756469.png)  | ![screen shot 2017-03-25 at 8 32 13 pm](https://cloud.githubusercontent.com/assets/378023/24321855/2fa0071c-119a-11e7-8d02-6991e391ee52.png)

Did the gap get added because the One themes had a gap between panes? In https://github.com/atom/one-light-ui/pull/96 I removed the "Spacious" mode, so now there are no more gaps and just `1px` borders. So I had to use `!important` to [override](https://github.com/atom/one-light-ui/pull/96/files#diff-f72770f52e9637dd2337d151b4d73e45R7) the `width/height`.

Was also thinking of adding a `1px` border on all panes and panels. Because currently packages do that but then you get double borders when two packages are next to each other. So if core would take care of borders, packages only have to think about what's "inside". Themes that don't like borders could remove them or just do `@base-border-color: transparent`.